com.obsproject.Studio.Plugin.obs-kinect.yaml

```diff
@@ -0,0 +1,78 @@
+finish-args:
+  - --env=LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/app/plugins/lib/obs-plugins
+  - --env=LIBFREENECT2_PIPELINE=cpu
+modules:
+  - shared-modules/libusb/libusb.json
+
+#  - name: glfw
+#    buildsystem: cmake
+#    config-opts:
+#      - -DBUILD_SHARED_LIBS=ON
+#    sources:
+#      - type: archive
+#        url: https://github.com/glfw/glfw/releases/download/3.3.8/glfw-3.3.8.zip
+#        sha256: 4d025083cc4a3dd1f91ab9b9ba4f5807193823e565a5bcf4be202669d9911ea6
+
+#  - name: libjpeg-turbo
+#    buildsystem: cmake
+#    config-opts:
+#      - -DCMAKE_INSTALL_LIBDIR=/app/plugins/obs-kinect/lib
+#    sources:
+#      - type: archive
+#        url: https://github.com/libjpeg-turbo/libjpeg-turbo/archive/refs/tags/2.1.4.tar.gz
+#        sha256: a78b05c0d8427a90eb5b4eb08af25309770c8379592bb0b8a863373128e6143f
+
+  - name: libfreenect
+    buildsystem: cmake
+    sources:
+      - type: archive
+        url: https://github.com/OpenKinect/libfreenect/archive/refs/tags/v0.6.4.tar.gz
+        sha256: 6169600f999729a7f99dd71a9825ed6e7aec8b8aac4f532ecf2911f76133c125
+
+  - name: libfreenect2
+    buildsystem: cmake
+    sources:
+      - type: archive
+        url: https://github.com/OpenKinect/libfreenect2/archive/refs/tags/v0.2.1.tar.gz
+        sha256: c09e52c97b0e90335f4762ed5363293ad0fc1ea0064e578b879e7d15cc3559df
+
+  - name: obs-kinect
+    buildsystem: simple
+    build-commands:
+      - unzip  ubuntu*.zip
+      - install -D obs-plugins/64bit/* ${FLATPAK_DEST}/lib/
+      - mkdir -p ${FLATPAK_DEST}/share/obs/obs-plugins/
+      - mv data/obs-plugins/obs-kinect/ ${FLATPAK_DEST}/share/obs/obs-plugins/
+    post-install:
+      - install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo ${FLATPAK_ID}.metainfo.xml
+      - appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}
+
+      ### Everything below here is dirty file manipulation to put libraries to /lib/obs-plugins
+
+      - rm -rf ${FLATPAK_DEST}/include ${FLATPAK_DEST}/bin ${FLATPAK_DEST}/lib/cmake/ ${FLATPAK_DEST}/lib/pkgconfig/ ${FLATPAK_DEST}/lib/fakenect/ ${FLATPAK_DEST}/lib/libfreenect_sync.a ${FLATPAK_DEST}/lib/libfreenect.a ${FLATPAK_DEST}/lib/libusb-1.0.la
+      - mkdir ${FLATPAK_DEST}/lib/obs-plugins
+      - cp -a ${FLATPAK_DEST}/lib/{lib*,obs-kinect*} ${FLATPAK_DEST}/lib/obs-plugins/
+#      - find ${FLATPAK_DEST}/lib/ -maxdepth 1 -type f -delete
+#      - find ${FLATPAK_DEST}/lib/ -maxdepth 1 -type l -delete
+      - mkdir -p ${FLATPAK_DEST}/usr/lib
+      - cp -a ${FLATPAK_DEST}/lib/{lib*,obs-kinect*} ${FLATPAK_DEST}/usr/lib/
+    sources:
+#      - type: archive
+#        url: https://github.com/SirLynix/obs-kinect/suites/10096917855/artifacts/492222746
+#        sha256: f9ad192d98e7c478d8cd57c0e21c321f099fc7f2d9e972c6528386b285b1d070
+      - type: file
+        path: com.obsproject.Studio.Plugin.obs-kinect.metainfo.xml
+      - type: file
+        path: ubuntu-20.04-x64-releasedbg.zip
```

Comment:
You shall not use pre-built binaries meant for Ubuntu but build the plugin from its source.

Reply:
I couldn't find a way to build xmake programs in flatpak. I've seen other complex apps skipping the building in flathub. Can't I do that same here?

Reply:
Maybe install xmake as a module if not present in the runtime, then use "simple" as buildsystem and put all the commands in `build-commands` and make sure to avoid having xmake to download dependencies while building.

> Can't I do that same here?

This method will never guarantee that the plugin will always work inside a Flatpak runtime.